Class Animal.

kram3ko opened the pull request:
Herbivore inherit  from Animal,
Carnivore from Animal. Completed logic of classes.

kram3ko committed:
Class Animal.  Herbivore inherit  from Animal,
Carnivore from Animal. Completed logic of classes.

oliasymak24 commented on app/main.py:
```diff
@@ -1 +1,38 @@
+    health = 100
+    hidden = False
+
+    def __init__(self, name: str, health: int = 100,
+                 hidden: bool = False) -> None:
```
Each parameter must be from new line

kram3ko replied:
used instead dataclass. But thanks understand.

oliasymak24 commented on app/main.py:
```diff
@@ -1 +1,38 @@
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore: Herbivore) -> Herbivore:
```
you don't need self here, so make this method static

kram3ko replied:
set it as staticmethod.

oliasymak24 commented on app/main.py:
```diff
@@ -1 +1,38 @@
+class Animal:
+    alive = []
+    health = 100
+    hidden = False
```
This is renudant, move to _init_

kram3ko replied:
removed duplicate code.. Understand mistake. Thanks

oliasymak24 commented on app/main.py:
```diff
@@ -1 +1,38 @@
+
+    def hide(self) -> bool:
+        self.hidden = not self.hidden
+        return self.hidden
```
No need for return

kram3ko committed:
For easy init used data class
Removed health, hidden from Class attributes.
in class Herbivore removed return in method hide
in class Carnivore made staticmethod bite and
if not isinstance just return None.

oliasymak24 commented on app/main.py:
```diff
@@ -1 +1,36 @@
+    health: int = 100
+    hidden: bool = False
+
+    # add instance to Class list(alive) after init
```
You need always clean up code and deleting comments

kram3ko replied:
Ok. Thank you. understand.
